x-pack/plugins/apm/server/lib/anomaly_detection/get_anomaly_detection_jobs.ts

```diff
@@ -5,56 +5,40 @@
  */
 
 import { Logger } from 'kibana/server';
-import { PromiseReturnType } from '../../../../observability/typings/common';
 import { Setup } from '../helpers/setup_request';
-import { AnomalyDetectionJobByEnv } from '../../../typings/anomaly_detection';
 import { ML_GROUP_NAME_APM } from './create_anomaly_detection_jobs';
+import { hasLegacyJobs } from './has_legacy_jobs';
 
-export type AnomalyDetectionJobsAPIResponse = PromiseReturnType<
-  typeof getAnomalyDetectionJobs
->;
-export async function getAnomalyDetectionJobs(
-  setup: Setup,
-  logger: Logger
-): Promise<AnomalyDetectionJobByEnv[]> {
+export async function getAnomalyDetectionJobs(setup: Setup, logger: Logger) {
   const { ml } = setup;
   if (!ml) {
-    return [];
+    return;
   }
-  try {
-    const mlCapabilities = await ml.mlSystem.mlCapabilities();
-    if (
-      !(
-        mlCapabilities.mlFeatureEnabledInSpace &&
-        mlCapabilities.isPlatinumOrTrialLicense
-      )
-    ) {
-      logger.warn(
-        'Anomaly detection integration is not availble for this user.'
-      );
-      return [];
-    }
-  } catch (error) {
-    logger.warn('Unable to get ML capabilities.');
-    logger.error(error);
-    return [];
-  }
-  try {
-    const { jobs } = await ml.anomalyDetectors.jobs(ML_GROUP_NAME_APM);
-    return jobs
-      .map((job) => {
-        const environment = job.custom_settings?.job_tags?.environment ?? '';
-        return {
-          job_id: job.job_id,
-          environment,
-        };
-      })
-      .filter((job) => job.environment);
-  } catch (error) {
-    if (error.statusCode !== 404) {
-      logger.warn('Unable to get APM ML jobs.');
-      logger.error(error);
-    }
-    return [];
+
+  const mlCapabilities = await ml.mlSystem.mlCapabilities();
+  if (
+    !(
+      mlCapabilities.mlFeatureEnabledInSpace &&
+      mlCapabilities.isPlatinumOrTrialLicense
+    )
+  ) {
+    logger.warn('Anomaly detection integration is not availble for this user.');
+    return;
   }
+
+  const { jobs } = await ml.anomalyDetectors.jobs(ML_GROUP_NAME_APM);
+  const apmJobs = jobs
+    .map((job) => {
+      const environment = job.custom_settings?.job_tags?.environment ?? '';
+      return {
+        job_id: job.job_id,
+        environment,
+      };
+    })
+    .filter((job) => job.environment);
+
+  return {
+    jobs: apmJobs,
+    hasLegacyJobs: await hasLegacyJobs(setup),
+  };
```

Comment:
@ogupte I removed the try/catch here. LMK if they are necessary. I think we can handle these errors on the client. 
If we happen to call this service from, say, the service map, we should try/catch at the call point.

```
let res;
try {
  res = await getAnomalyDetectionJobs()
} catch (e) {...}